.github/workflows/docker_image_build.yaml

```
@@ -1,22 +1,48 @@
-name: Building the Docker image containing the basic tools for building ParallelZone
+# Docker images are re-built only when the corrsponding docker files under Docker/ are changed
+name: Building the Docker images containing the basic tools for building repos
```

I'd recommend shorter names. They'll appear on the GitHub dashboards and in the PR status. If they're too long they'll cut off anyways.
```suggestion
name: Build base Docker images
```